<a href="https://colab.research.google.com/github/Chandradithya8/Machine_learning-Projects/blob/master/iris_model1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [13]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_iris
from statistics import mean

In [14]:
from sklearn.model_selection import cross_val_score

In [15]:
iris=load_iris()
dir(iris)

['DESCR', 'data', 'feature_names', 'filename', 'target', 'target_names']

In [16]:
l=cross_val_score(LogisticRegression(),iris.data,iris.target,cv=3)
mean(l)

/usr/local/lib/python3.6/dist-packages/sklearn/linear_model/_logistic.py:940: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  extra_warning_msg=_LOGISTIC_SOLVER_CONVERGENCE_MSG)


0.9733333333333333

In [17]:
s=cross_val_score(SVC(),iris.data,iris.target,cv=3)
mean(s)

0.96

In [18]:
d=cross_val_score(DecisionTreeClassifier(),iris.data,iris.target,cv=3)
mean(d)

0.9733333333333333

In [19]:
r=cross_val_score(RandomForestClassifier(n_estimators=100),iris.data,iris.target,cv=3)
mean(r)

0.96

# FROM THIS WE GET THE BEST ACCURACY FROM LOGISTIC REGRESSION AND DECISION TREE CLASSIFIER : 0.9733333333333333